ci/pipelines/build-and-release.yaml

```diff
@@ -0,0 +1,145 @@
+# Pipeline to build and release oauth2_proxy
+# Based on https://github.com/vito/booklit/blob/master/ci/pipeline.yml
+
+groups:
+  - name: dev
+    jobs: [tests, rc, ship]
+
+  - name: semver
+    jobs: [major, minor, patch]
+
+resource_types:
+  - name: github-status
+    type: registry-image
+    source:
+      repository: resource/github-status
+
+resources:
+  - name: oauth2_proxy
+    type: git
+    icon: github-circle
+    source:
+      uri: git@github.com:states-of-guernsey/oauth2_proxy.git
+      branch: 4.0.0-fix
+      private_key: ((github_private_key))
+      ignore_paths:
+        - ci/
+
+  - name: ci
+    type: git
+    icon: github-circle
+    source:
+      uri: git@github.com:states-of-guernsey/oauth2_proxy.git
+      branch: 4.0.0-fix
+      private_key: ((github_private_key))
+      paths:
+        - ci/
+
+  - name: gh-status
+    type: github-status
+    icon: bug-check-outline
+    source:
+      repo: states-of-guernsey/oauth2_proxy
+      access_token: ((github_access_token))
+
+  - name: version
+    type: semver
+    icon: tag
+    source:
+      region_name: eu-west-1
+      bucket: ((build_artifacts_bucket.bucket_name))
+      access_key_id: ((build_artifacts_bucket.access_key_id))
+      secret_access_key: ((build_artifacts_bucket.secret_access_key))
+      initial_version: 4.0.0
```

Comment:
I'd bump this to 5.0.0. This is only ever used if the semver resource isn't already populated, which it is, but just being careful